test/VaultRegistry.test.ts

```diff
@@ -1,445 +1,943 @@
+                            now() -
+                            Number(this.governanceDelay)
+                    )
+                ).to.be.lessThanOrEqual(1);
+            });
+
+            describe("access control:", () => {
+                it("allowed: any address", async () => {
+                    await withSigner(randomAddress(), async (s) => {
+                        await expect(
+                            this.subject.stagedProtocolGovernanceTimestamp()
+                        ).to.not.be.reverted;
+                    });
+                });
+            });
+
+            describe("edge cases", () => {
+                describe("when nothing is staged", () => {
+                    it("returns 0", async () => {
+                        expect(
+                            await this.subject.stagedProtocolGovernanceTimestamp()
+                        ).to.be.equal(0);
+                    });
+                });
+                describe("right after #commitStagedProtocolGovernance was called", () => {
+                    it("returns 0", async () => {
+                        await this.subject
+                            .connect(this.admin)
+                            .stageProtocolGovernance(
+                                this.newProtocolGovernance.address
+                            );
+                        await sleep(this.governanceDelay);
+                        await this.subject
+                            .connect(this.admin)
+                            .commitStagedProtocolGovernance();
+                        expect(
+                            await this.subject.stagedProtocolGovernanceTimestamp()
+                        ).to.be.equal(0);
+                    });
+                });
+            });
+        });
+
+        describe("#vaultsCount", () => {
+            it("returns the number of registered vaults", async () => {
+                expect(await this.subject.vaultsCount()).to.be.equal(1);
+            });
+
+            describe("access control:", () => {
+                it("allowed: any address", async () => {
+                    await withSigner(randomAddress(), async (s) => {
+                        await expect(this.subject.connect(s).vaultsCount()).to
+                            .not.be.reverted;
+                    });
+                });
+            });
+
+            describe("properties", () => {
+                pit(
+                    "@property: when N new vaults have been registered, vaults count will be increased by N",
+                    { numRuns: RUNS.verylow },
+                    integer({ min: 1, max: 5 }),
+                    async (vaultsCount: number): Promise<boolean> => {
+                        let oldVaultsCount = Number(
+                            await this.subject.vaultsCount()
+                        );
+                        for (var i = 0; i < vaultsCount; ++i) {
+                            await deployments.deploy("MockERC165", {
```

Comment:
deployment takes a considerable time. I'd make a fixture here of deploying 5 Mock contracts. Inside the cycle just call registerVault